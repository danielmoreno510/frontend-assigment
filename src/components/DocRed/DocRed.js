import React from "react";
import PropTypes from "prop-types";

import Header from "../Header";
import MainArticle from "../MainArticle";
import ArticleList from "../ArticleList";
import Footer from "../Footer";
import styles from "./styles";

const HomePage = ({ articles }) => {
  return (
    <div>
      <Header />
      <div style={styles.container}>
        <MainArticle article={articles[0]} />
        <ArticleList articles={articles} />
      </div>
      <Footer />
    </div>
  );
};

HomePage.propTypes = {
  articles: PropTypes.array.isRequired,
};

export default HomePage;
