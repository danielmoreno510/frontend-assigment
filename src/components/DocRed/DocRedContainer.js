import DocRed from "./DocRed";
import { StoreService } from "../../state/StoreService.js";

const mapStateToProps = (state) => ({
  articles: state.articles.articleList,
});

const HomePageContainer = StoreService.connect(mapStateToProps)(DocRed);
export default HomePageContainer;
