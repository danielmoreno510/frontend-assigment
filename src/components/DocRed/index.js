import DocRed from './DocRedContainer'

export default DocRed