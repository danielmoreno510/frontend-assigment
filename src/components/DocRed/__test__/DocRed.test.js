import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";

import DocRed from "../DocRed";

configure({ adapter: new Adapter() });
const testProps = {
  articles: [],
};

describe("DocRed component", () => {
  describe("WHEN the component is render", () => {
    const wrapper = shallow(<DocRed {...testProps} />);

    it("THEN should display DocRed component", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
