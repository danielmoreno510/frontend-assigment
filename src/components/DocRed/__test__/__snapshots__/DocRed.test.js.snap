// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocRed component WHEN the component is render THEN should display DocRed component 1`] = `
<div>
  <Header />
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <MainArticle />
    <ArticleList
      articles={Array []}
    />
  </div>
  <Footer />
</div>
`;
